ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/ImportOrderingRule.kt

```diff
@@ -147,6 +147,37 @@ public class ImportOrderingRule :
         ideaImportsLayoutProperty
     )
 
+    private fun getUniqueImportsAndBlankLines(
+        children: Array<ASTNode>,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ): Pair<Boolean, List<ASTNode>> {
+        var autoCorrectDuplicateImports = false
+        val imports = mutableListOf<ASTNode>()
+        val importTextSet = mutableSetOf<String>()
+
+        children.forEach { current ->
+            val isPsiWhiteSpace = current.psi is PsiWhiteSpace
+
+            if (
+                current.elementType == ElementType.IMPORT_DIRECTIVE ||
+                isPsiWhiteSpace && current.textLength > 1 // also collect empty lines, that are represented as "\n\n"
+            ) {
+                if (isPsiWhiteSpace || importTextSet.add(current.text)) {
```

Comment:
These lines are the most significant improvements in this PR.
The existing code was inefficient at checking for duplicate `imports`, but this change reduces the amount of computation.
(However, this is generally only a minor improvement, as the use case of thousands of `import` lines in a single file is unlikely.)

Reply:
The performance improvement is neglible. I do like the improved readability.